_includes/google_analytics.html

```diff
@@ -0,0 +1,8 @@
+<script async src="https://www.googletagmanager.com/gtag/js?id={{ include.ga_property }}"></script>
+<script>
+  window.dataLayer = window.dataLayer || [];
+  function gtag(){dataLayer.push(arguments);}
+  gtag('js', new Date());
+
+  gtag('config', '{{ include.ga_property }}'{% unless site.ga_tracking_anonymize_ip == nil %}, { 'anonymize_ip': true }{% endunless %});
```

Comment:
What if we have the for loop go over this? i.e.

```suggestion
{% for ga_property in site.ga_tracking %}
  gtag('config', '{{ ga_property }}'{% unless site.ga_tracking_anonymize_ip == nil %}, { 'anonymize_ip': true }{% endunless %});
  {% endfor %}
```

(and do the relevant else block to deal with it being just a string; this is not a complete suggestion)

This way, you're only including the script once. In your current code, it seems like you're double-pushing the date, and redefining `gtag`/`dataLayer` twice, which is unnecessary.

Reply:
Hmm... Yeah, I'll need to do some testing on this to ensure the Google Analytics tracking isn't oddly affected, but you're right. I think I'll just make the loop for the property-related code and keep those initial definitions in their own, singular instance. Because the first line should also be in the loop, and I'd rather not have two identical loops. It'd be something like this:
```liquid
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
</script>
{% for ga_property in site.ga_tracking %}
    <script async src="https://www.googletagmanager.com/gtag/js?id={{ include.ga_property }}"></script>
    <script>gtag('config', '{{ include.ga_property }}'{% unless site.ga_tracking_anonymize_ip == nil %}, { 'anonymize_ip': true }{% endunless %});</script>
{% endfor %}
```

Reply:
Ah, I totally missed that it's also in the tag URL. Hm. I'm curious why the query string is needed to be encoded if it's also part of the datalayer.

If you need to load two scripts anyways, what you have in the PR right now is totally fine. Let me think on this after NeurIPS outreach is done (i.e. tomorrow); if you find some definitive documentation indicating one way or another though, do let me know!